csv-data-management/extract_data_from_csv.py

```diff
@@ -200,6 +200,119 @@ def merge_two_csv_and_extract_data(config, params):
+            columnNames = columnNames.split(",")
+            
+            # We are passing  specific columns name to filter data from here
+            df_file1 =  _read_file_specific_columns(first_file_path,columnNames,numberOfRowsToSkipFirst)
+            
+        elif isCSVWithoutHeadersFirst: # CSV file without column header and more than one column
+            df_file1 =  _read_file_no_headers(first_file_path,numberOfRowsToSkipFirst) 
+
+        elif isSingleColumnFrist: #CSV with single column and header    
+            df_file1 = _read_file_single_column(first_file_path,numberOfRowsToSkipFirst)
+        
+        elif isSingleColumnFrist and not isCSVWithoutHeadersFirst: # CSV file with one  column and no header 
+            df_file1 = _read_file_single_column_no_header(first_file_path,numberOfRowsToSkipFirst)
+
+        else:  
+            # We are reading complete file assuming it has column header
+            df_file1 = _read_file_all_columns(first_file_path,numberOfRowsToSkipFirst)
+
+        #Lets read second file 
+        
+        if params.get('numberOfRowsToSkipSecond'):
+            numberOfRowsToSkipSecond = params.get('numberOfRowsToSkipSecond')
+          
+        if params.get('file2_columnNames') != "":  # CSV file with column header and specifci columns to use in creating recordset 
+            columnNames = params.get('file2_columnNames')
+            columnNames = columnNames.split(",")
+    
+            # We are passing  specific columns name to filter data from here
+            df_file2 = _read_file_specific_columns(second_file_path,columnNames,numberOfRowsToSkipSecond)
+            
+        elif isCSVWithoutHeadersSecond: # CSV file without column header and more than one column
+            df_file2 = _read_file_no_headers(second_file_path,numberOfRowsToSkipSecond)  
+
+        elif isSingleColumnSecond: #CSV with single colum   
+            df_file2 = _read_file_single_column(second_file_path,numberOfRowsToSkipSecond)  
+
+        elif isSingleColumnSecond and  isCSVWithoutHeadersSecond: # CSV file with one  column and no header 
+            df_file2 = _read_file_single_column_no_header(second_file_path,numberOfRowsToSkipSecond) 
+
+        else:  
+            # We are reading complete file assuming it has column header
+            df_file2 = _read_file_all_columns(second_file_path,numberOfRowsToSkipSecond)
+
+        
+        #Merge both files
+        combined_recordSet =pd.concat([df_file1,df_file2])    
+
+        # If user has selected to dedeuplicate recordset
+        try:
+            if params.get('deDupValuesOn'):
+                deDupValuesOn = params.get('deDupValuesOn')
+                deDupValuesOn = deDupValuesOn.split(",")
+                combined_recordSet.drop_duplicates(subset=deDupValuesOn, keep='first')
+        except Exception as Err:
+            logger.error('Error in dedeuplicating data  extract_data_from_csv(): %s' % Err)
```

Comment:
function name will be present in stack trace.

Reply:
but, why we are still continuing ahead with exception?

Reply:
Statements in logs are more clearer , and we can search better as single connector log is used to write for all connectors 

With exception thrown code terminates , did not see it continuing , but will re-test